src/test/java/com/googlecode/lanterna/issue/Issue595.java

```diff
@@ -18,46 +18,91 @@
-                        }
-                    }, executorService)
-                    .exceptionally(e -> {
-                        throw new RuntimeException(e);
-                    });
+                                 try {
+                                     Thread.sleep(SLEEP_MILLIS);
+                                 } catch (final InterruptedException e) {
+                                     Thread.currentThread()
+                                           .interrupt();
+                                     throw new RuntimeException(e);
+                                 } finally {
+                                     waitingDialog.close();
+                                 }
+                             }, executorService)
+                             .exceptionally(e -> {
+                                 throw new RuntimeException(e);
+                             });
             waitingDialog.waitUntilClosed();
-            System.out.println("WAIT DIALOG CLOSED");
-        } catch (final IOException e) {
-            throw new RuntimeException(e);
+
+            // Ensure Executor Thread Dead
+            executorService.shutdownNow();
+            if (!executorService.awaitTermination(SLEEP_SECONDS, TimeUnit.SECONDS)) {
+                throw new IllegalStateException("ExecutorService Unstoppable");
+            }
+
+            // Check for Animated Label Hanging Thread
+            final String animatedLabelName = AnimatedLabel.class.getSimpleName()
+                                                                .toLowerCase();
+            final Optional<Thread> optionalAnimatedLabelThread = Thread.getAllStackTraces()
+                                                                       .keySet()
+                                                                       .stream()
+                                                                       .filter(thread -> thread.getName()
+                                                                                               .toLowerCase()
+                                                                                               .contains(animatedLabelName))
+                                                                       .findAny();
+            if (!optionalAnimatedLabelThread.isPresent()) {
+                return;
+            }
+            final Thread thread = optionalAnimatedLabelThread.get();
+            thread.join(SLEEP_MILLIS);
+            if (thread.isAlive()) {
+                throw new IllegalStateException("AnimatedLabel Thread Waiting");
+            }
+
+        } catch (final Throwable e) {
+            if (e instanceof InterruptedException) {
+                Thread.currentThread()
+                      .interrupt();
+            }
+            ++exit_code;
+            final StringWriter stringWriter = new StringWriter();
+            e.printStackTrace(new PrintWriter(stringWriter));
```

Comment:
redundant, can just call `e.printStackTrace()`

Reply:
Things like `e.printStackTrace()` are often barfed at by code quality tools, like e.g. SonarQube.
The preferred way would be resorting to some logging-framework, so that it can be disabled
for production use - and not dump an exception-stackdump into user's face.

I'm aware that using a logging framework is not a simple decision, but the original code is already closer to a real solution than "simplifying" it back to `e.printStackTrace()`.

My opinion: just leave that bit as it is.  Simplifying it as suggested would be a step in the wrong direction.

Reply:
Unless we resort to a logging framework

```java
StringWriter s = new StringWriter();
e.printStackTrace(new PrintWriter(s));
System.err.print(s);
```

Is equivalent to `e.printStackTrace()` [(at least in vanilla JDK implementations)](https://github.com/openjdk/jdk/blob/998d0baab0fd051c38d9fd6021628eb863b80554/src/java.base/share/classes/java/lang/Throwable.java#L677-L679)

Reply:
This was copy+pasted from the bug report, I think :)

Reply:
Nevermind, wrong PR